### ISSUE TITLE
Question about the factor of compensation in dropout

### ISSUE THREAD

**Issue by yzhangcs** [jcyk/Dynet-Biaffine-dependency-parser#10]
https://github.com/jcyk/Dynet-Biaffine-dependency-parser/blob/50b287bcb4a7decd1faa461c54d9c5b3fb348ee5/models/baseParser.py#L59
I guess this operation is the simplification of the source code, right?
But I think it is not strictly equivalent to what is described in the paper:
> We drop 33% of words and 33% of tags during training: when one is dropped the other is scaled by a factor of two to compensate, and when both are dropped together, the model simply gets an input of zeros.

I just modified it and it seems that the following one is better:
```
scale = 2. / (word_mask + tag_mask + 1e-12) 
```
Am I right?

**Comment by jcyk** [comment]
Hi, the description in the paper is not precise. My previous line is
consistent with their released code.
Note for word, it uses a sum of pretrained and random initialized
embeddings. So actually word counts 2.

**Comment by yzhangcs** [comment]
Yeah, you solved my puzzle.
But I saw the author's [another paper](https://web.stanford.edu/~tdozat/files/TDozat-CoNLL2017-Paper.pdf), which uses the same way to do the dropout, in his released code, the method is similar to what I said:
https://github.com/tdozat/Parser-v2/blob/304c638aa780a5591648ef27060cfa7e4bee2bd0/parser/neural/models/nn.py#L69-L70
And in [Parser-v3](https://github.com/tdozat/Parser-v3), he gave up the compensation directly.
I don't know why.
